lms/security.py

```diff
@@ -212,13 +213,24 @@ def _get_lti_jwt(request):
     if not id_token:
         return {}
 
-    try:
-        jwt_params = jwt.decode(id_token, options={"verify_signature": False})
-    except InvalidTokenError as err:
-        raise marshmallow.ValidationError("Invalid id_token", "authorization") from err
+    jwt_service = request.find_service(JWTService)
+    registration_service = request.find_service(LTIRegistrationService)
 
-    warnings.warn("Using not verified JWT token")
-    return jwt_params
+    # Get the target registration of the JWT.
+    # We'll use the public key URL in the registration to verify the token
+    unverified_payload = jwt_service.decode_unverified(id_token)
```

Comment:
There's a circular dependency between the contents of the JWT and the info required to verify it.

The JWT library itself does something like this at some point so I think it's fine.